### PR TITLE
Add update params to `marker` module

### PR DESCRIPTION
**Pull request by nullpointer0x00**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

## Summary by CodeRabbit

- **New Features**
  - Added `UpdateParams` endpoint and CLI for the marker module.
  - Added `EventMarkerParamsUpdated` event to track marker parameter updates.

- **Bug Fixes**
  - Fixed validation logic for supply increase by using `GetMaxSupply` instead of `GetParams(ctx).MaxSupply`.

- **Documentation**
  - Updated Swagger documentation to include `MsgUpdateParamsResponse`.

- **Tests**
  - Added comprehensive test cases for `UpdateParams` endpoint and marker parameter updates.
  - Enhanced validation tests for `Params` struct.

- **Refactor**
  - Consolidated logic for retrieving maximum supply into `GetMaxSupply`.
  - Removed deprecated `GetMaxTotalSupply` function.

### CODE REVIEW

File: x/marker/client/cli/cli_test.go [hunk 1]
```diff
@@ -2552,3 +2552,62 @@ func (s *IntegrationTestSuite) TestUpdateParamsProposal() {
...
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastSync),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewInt64Coin(s.cfg.BondDenom, 10)).String()),
+			)
+			testcli.NewTxExecutor(tc.cmd, tc.args).
+				WithExpErrMsg(tc.expectErr).
+				WithExpCode(tc.expectedCode).
+				Execute(s.T(), s.testnet)
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (nullpointer0x00) [reply x/marker/client/cli/cli_test.go:2613]: Good catch!

File: x/marker/client/cli/tx.go [hunk 2]
```diff
@@ -95,6 +96,7 @@ func NewTxCmd() *cobra.Command {
 		GetCmdRemoveAdministratorProposal(),
 		GetCmdChangeStatusProposal(),
 		GetCmdWithdrawEscrowProposal(),
+		GetCmdMarkerProposal(),
```
[the remark this answers is not in the document]
Comment (nullpointer0x00) [reply x/marker/client/cli/tx.go:99]: Intended to add `GetUpdateMarkerParamsCmd()`.

File: x/marker/keeper/params_test.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ func TestParamTestSuite(t *testing.T) {
 
 func (s *ParamTestSuite) TestGetSetParams() {
 	defaultParams := s.app.MarkerKeeper.GetParams(s.ctx)
-	s.Require().Equal(types.DefaultMaxTotalSupply, defaultParams.MaxTotalSupply, "Default MaxTotalSupply should match")
+	s.Require().Equal(uint64(0), defaultParams.MaxTotalSupply, "MaxTotalSupply is deprecated and should not be set")
```
[the remark this answers is not in the document]
Comment (nullpointer0x00) [reply x/marker/keeper/params_test.go:36]: This was checked to make sure the deprecated value wasn't being set by the previous function all.

**Comment by Taztingo** [comment]
LGTM, but you probably want to update the spec with the new included Event.

**Review by Taztingo — APPROVED** [review]

File: proto/provenance/marker/v1/marker.proto [hunk 4]
```diff
@@ -208,3 +208,10 @@ message EventSetNetAssetValue {
   string volume = 3;
   string source = 4;
 }
+
+// EventMarkerParamsUpdated event emitted when marker params are updated.
+message EventMarkerParamsUpdated {
```
Comment (Taztingo) [inline proto/provenance/marker/v1/marker.proto:213]: Spec should be updated to include this Event.

**Review by iramiller — DISMISSED** [review]

File: proto/provenance/marker/v1/marker.proto (hunk 4 above)
Comment (nullpointer0x00) [reply proto/provenance/marker/v1/marker.proto:213]: Yep, good catch! Done.

**Review by Taztingo — APPROVED** [review]

**Review by iramiller — APPROVED** [review]

**Pull request merged**